File: maps/src/main/java/it/localhostsoftware/maps/google/model/GoogleMarkerOptions.java
package it.localhostsoftware.maps.google.model;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import it.localhostsoftware.maps.model.BitmapDescriptor;
import it.localhostsoftware.maps.model.LatLng;
import it.localhostsoftware.maps.model.MarkerOptions;

public class GoogleMarkerOptions implements MarkerOptions {
    private final com.google.android.gms.maps.model.MarkerOptions markerOptions;

    public GoogleMarkerOptions() {
        this.markerOptions = new com.google.android.gms.maps.model.MarkerOptions();
    }

    public com.google.android.gms.maps.model.MarkerOptions getMarkerOptions() {
        return markerOptions;
    }

    @Override
    public MarkerOptions position(@NonNull LatLng<?> latLng) {
        markerOptions.position((com.google.android.gms.maps.model.LatLng) latLng.getLatLng());
        return this;
    }

    @Override
    public MarkerOptions zIndex(float v) {
        markerOptions.zIndex(v);
        return this;
    }

    @Override
    public MarkerOptions icon(@Nullable BitmapDescriptor<?> var1) {
        markerOptions.icon((com.google.android.gms.maps.model.BitmapDescriptor) var1.getBitmapDescriptor());
        return this;
    }

    @Override
    public MarkerOptions anchor(float v, float v1) {
        markerOptions.anchor(v, v1);
        return this;
    }

    @Override
    public MarkerOptions infoWindowAnchor(float v, float v1) {
        markerOptions.infoWindowAnchor(v, v1);
        return this;
    }

    @Override
    public MarkerOptions title(@Nullable String s) {
        markerOptions.title(s);
        return this;
    }

    @Override
    public MarkerOptions snippet(@Nullable String s) {
        markerOptions.snippet(s);
        return this;
    }

    @Override
    public MarkerOptions draggable(boolean b) {
        markerOptions.draggable(b);
        return this;
    }

    @Override
    public MarkerOptions visible(boolean b) {
        markerOptions.visible(b);
        return this;
    }

    @Override
    public MarkerOptions flat(boolean b) {
        markerOptions.flat(b);
        return this;
    }

    @Override
    public MarkerOptions rotation(float v) {
        markerOptions.rotation(v);
        return this;
    }

    @Override
    public MarkerOptions alpha(float v) {
        markerOptions.alpha(v);
        return this;
    }

    @Override
    public LatLng<?> getPosition() {
        return new GoogleLatLng(markerOptions.getPosition());
    }

    @Override
    public String getTitle() {
        return markerOptions.getTitle();
    }

    @Override
    public String getSnippet() {
        return markerOptions.getSnippet();
    }

    @Override
    public BitmapDescriptor<?> getIcon() {
        return new BitmapDescriptor<>(markerOptions.getIcon());
    }

    @Override
    public float getAnchorU() {
        return markerOptions.getAnchorU();
    }

    @Override
    public float getAnchorV() {
        return markerOptions.getAnchorV();
    }

    @Override
    public boolean isDraggable() {
        return markerOptions.isDraggable();
    }

    @Override
    public boolean isVisible() {
        return markerOptions.isVisible();
    }

    @Override
    public boolean isFlat() {
        return markerOptions.isFlat();
    }

    @Override
    public float getRotation() {
        return markerOptions.getRotation();
    }

    @Override
    public float getInfoWindowAnchorU() {
        return markerOptions.getInfoWindowAnchorU();
    }

    @Override
    public float getInfoWindowAnchorV() {
        return markerOptions.getInfoWindowAnchorV();
    }

    @Override
    public float getAlpha() {
        return markerOptions.getAlpha();
    }

    @Override
    public float getZIndex() {
        return markerOptions.getZIndex();
    }
}

File: maps/src/main/java/it/localhostsoftware/maps/google/model/GoogleMarkerOptions.kt
package it.localhostsoftware.maps.google.model

import com.google.android.gms.maps.model.MarkerOptions
import it.localhostsoftware.maps.model.BitmapDescriptor
import it.localhostsoftware.maps.model.LatLng

class GoogleMarkerOptions(markerOptions: MarkerOptions) : it.localhostsoftware.maps.model.MarkerOptions<MarkerOptions>(markerOptions) {
    override var position: LatLng<*>
        get() = GoogleLatLng(mo.position)
        set(value) {
            mo.position(value.ll as com.google.android.gms.maps.model.LatLng)
        }
    override var title: String?
        get() = mo.title
        set(value) {
            mo.title(value)
        }
    override var snippet: String?
        get() = mo.snippet
        set(value) {
            mo.snippet(value)
        }
    override var icon: BitmapDescriptor<*>?
        get() = BitmapDescriptor(mo.icon)
        set(value) {
            mo.icon(value?.bitmapDescriptor as? com.google.android.gms.maps.model.BitmapDescriptor)
        }
    override var isDraggable: Boolean
        get() = mo.isDraggable
        set(value) {
            mo.draggable(value)
        }
    override var isVisible: Boolean
        get() = mo.isVisible
        set(value) {
            mo.visible(value)
        }
    override var isFlat: Boolean
        get() = mo.isFlat
        set(value) {
            mo.flat(value)
        }
    override var rotation: Float
        get() = mo.rotation
        set(value) {
            mo.rotation(value)
        }
    override var alpha: Float
        get() = mo.alpha
        set(value) {
            mo.alpha(value)
        }
    override var zIndex: Float
        get() = mo.zIndex
        set(value) {
            mo.zIndex(value)
        }
    override var anchor: Pair<Float, Float>
        get() = Pair(mo.anchorU, mo.anchorV)
        set(value) {
            mo.anchor(value.first, value.second)
        }
    override var infoWindowAnchor: Pair<Float, Float>
        get() = Pair(mo.infoWindowAnchorU, mo.infoWindowAnchorV)
        set(value) {
            mo.infoWindowAnchor(value.first, value.second)
        }
}